crates/swc_ecma_transforms_module/tests/fixture/common/strict/export-3/output.amd.js

```diff
@@ -0,0 +1,6 @@
+define([
+    "require",
+    "foo"
+], function(require, _foo) {
+    "use strict";
```

Comment:
Isn't this module?
We don't have `__esModule` for empty modules?

Reply:
OK, I checked the Babel output and find it has `__esModule`.
The conclusion is that if there is an any export statement, empty binding or not, it should output the `__esModule`, right?
I will fix this soon.

Reply:
I think so?
`tsc` also emits it. https://www.typescriptlang.org/play?module=1#code/KYDwDg9gTgLgBAbwL5A

Reply:
If there is nothing to export, but some import statements, tsc and babel behave differently.

I tested some examples of importing empty files in native ESM.
### I think as long as it's an esm module it should be always marked as `__esModule`.


In native ESM module environment, we cannot import default or named binding from empty module file.
An error will be thrown out.

We should mimic this behaviour when transfroming ESM module.
If there is no `__esModule ` mark, the imported default will be empty object `{}` which is not expected.

With `__esModule ` mark, we will get undefined instead.
It's not as perfect as native esm error report but better than empty object.

Reply:
Yeah I agree, we should have `__esModule`

Reply:
Is it possible for a non-esm to be passed into the cjs pass?

Reply:
I don't think so, because it's not a module and `cjs` pass is for modules